boefjes/app.py

```diff
@@ -84,28 +85,32 @@ def start_working(
         for queue in queues:
             try:
                 logger.info(f"Popping from queue {queue.id}")
-                task = scheduler_client.pop_task(queue.id)
+                p_item = scheduler_client.pop_item(queue.id)
             except (RequestHTTPError, HTTPError, ConnectionError):
                 logger.exception("Popping task from scheduler failed")
                 time.sleep(10 * settings.poll_interval)
                 continue
-            except Exception:
+            except:  # noqa
                 logger.exception("Popping task from scheduler failed")
                 time.sleep(10 * settings.poll_interval)
                 continue
 
-            if not task:
+            if not p_item:
                 logger.info(f"Queue {queue.id} empty")
                 continue
 
             try:
-                logger.info(f"Handling task[{task.data.id}]")
-                item_handler.handle(task.data)
+                logger.info(f"Handling task[{p_item.data.id}]")
+                item_handler.handle(p_item.data)
+                scheduler_client.patch_task(str(p_item.id), TaskStatus.COMPLETED)
```

Comment:
I think it would be best if the try-except is only around `item_handler.handle(p_item.data)`. The logging can be moved above the `try:` and the `patch_call` should be in an `else:` clause so that we only log and report the task as failed when the task itself throws an exception.

And we need to handle exceptions from calling `patch_task`.